packages/berry-pnpify/sources/NodePathResolver.ts

```diff
@@ -78,8 +81,32 @@ export class NodePathResolver {
     this.options = options;
   }
 
-  private getIssuer(pnp: PnpApi, pathname: string): string | undefined {
-    const locator = pnp.findPackageLocator(pathname + '/');
+  /**
+   * Returns `readdir`-like result for partially resolved pnp path
+   *
+   * @param issuerInfo issuer package information
+   * @param request either '' or '@scope'
+   *
+   * @returns `undefined` - if dir does not exist, or `readdir`-like list of subdirs in the virtual dir
+   */
+  public readDir(issuerInfo: PackageInformation, request: string): string[] | undefined {
+    const result = [];
+    for (const key of issuerInfo.packageDependencies.keys()) {
+      const [ scope, pkgName ] = key.split('/');
```

Comment:
You can use `structUtils.parseIdent`

Reply:
What is the point of doing so? `structUtils.parseIdent` has more complicated logic then `split('/')`, which is not needed here.

Reply:
Mostly to keep the logic in one place 🤔

It's less efficient from a perf perspective but it also contributes to lower the entry level to understand the code: readers can reuse the knowledge they already have of the `structUtils` functions to quickly figure out what happens. Otherwise it requires a bit more attention to figure out what happens.

A last reason is that it also ensures that the values have the expected format, so it fails early if something looks weird (for example if we somehow end up with `a/b/c` in the PnP map).

Reply:
That said since you don't use `@berry/core` yet there's an argument to be made that it could increase the bundle size for what is essentially a `split` ... 🤷

Reply:
It adds 50k to the bundle ... Webpack doesn't seem able to remove the unused symbols from `structUtils`, which in turn cause `semver` to be bundled 😕

Reply:
@arcanis Lets postpone this change for later. The code is less performant and not much more readable. Also I run into various troubles with compilation inside `berry-core` and I don't want to change `berry-core` in this PR.

Reply:
Ok 👍